packages/bubblewrap/swap-getcwd-func.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/bubblewrap.c b/bubblewrap.c
+index b17ff99..24b31ef 100644
+--- a/bubblewrap.c
++++ b/bubblewrap.c
+@@ -2963,7 +2963,14 @@ main (int    argc,
+   if (mount ("tmpfs", base_path, "tmpfs", MS_NODEV | MS_NOSUID, NULL) != 0)
+     die_with_error ("Failed to mount tmpfs");
+ 
+-  old_cwd = get_current_dir_name ();
++  /* get_current_dir_name() is not included in Bionic (Android's libc) */
++  /* old_cwd = get_current_dir_name (); */
++  old_cwd = getcwd(NULL, 0);
++  if (old_cwd == NULL)
++    {
++      die_with_error("failed in getting current working directory. " \
```

Comment:
The `\` is unnecessary. So is the `{}`.

Reply:
- I used \ for 80 char limit on line width.
- Used the braces in case anyone wanted to add stuff below it. I've read somewhere it's better to use block for `if`.
Shall I remove them?

Reply:
By `unnecessary` I wasn't referring to the break itself, but that `\` is unnecessary in C for the break.

There's no right or wrong about the `{}`. It's just that the code style adopted in bubblewrap does not use `{}` when there's just one *statement* (NOT the same thing as *line*), but it probably won't prevent others from merging this anyway. (While I helped to get this work, I myself do not encourage new package that may not have a use case to most / be usable on most's devices to be added. Therefore I'll let others decide whether this should be merged.)

Reply:
Thanks, I removed \ and {} .